Transactional Outbox documentation

jeroenkoknl opened the pull request:
Added documentation for the transactional outbox implementation in Rebus.SqlServer. This includes setup instructions, usage scenarios both inside and outside of Rebus handlers, and a high-level explanation of message processing after storing messages in the outbox. 

---
Rebus is [MIT-licensed](https://opensource.org/licenses/MIT). The code submitted in this pull request needs to carry the MIT license too. By leaving this text in, __I hereby acknowledge that the code submitted in the pull request has the MIT license and can be merged with the Rebus codebase__.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rebus-org/Rebus.SqlServer?pullRequest=110) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rebus-org/Rebus.SqlServer?pullRequest=110) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rebus-org/Rebus.SqlServer?pullRequest=110) it.</sub>

jeroenkoknl commented:
I don't understand why the AppVeyor build failed. This PR doesn't change any source code.

jeroenkoknl commented:
Hi there! 👋

I just wanted to kindly follow up on this PR as it's been over six weeks since it was submitted. I understand maintainers often have a lot on their plate, so I completely understand if it's just a matter of time or priority.

If there's anything I can do to help move this forward—like clarifying any details or addressing concerns—please let me know. Thanks again for maintaining this great library, and I really appreciate your work! 😊

mookid8000 commented:
Somehow I missed your PR – sorry!

Thanks for contributing!
